p2p/src/network/kad/p2p_network_kad_internals.rs

```diff
@@ -561,6 +571,15 @@ impl<const K: usize> P2pNetworkKadBucket<K> {
             }
 
             for addr in entry.addrs {
+                if e.addrs.len() == P2pNetworkKadEntry::MAX_ADDRS {
```

Comment:
@tizoc I truncate the list passed to `new` and the other place that loop that uses `push`. But I'll make that change to `>=` because it prevents accidental modifications. Also making the structure private as @vlad9486 suggests.